src/color.rs

```diff
@@ -104,38 +138,47 @@ pub enum ExtendedColorType {
 }
 
 impl ExtendedColorType {
-    /// Get the number of channels for colors of this type.
+    /// Get the type of channels for colors of this type.
     ///
-    /// Note that the `Unknown` variant returns a value of `1` since pixels can only be treated as
+    /// Note that the `Unknown` variant returns a luminance channel since pixels can only be treated as
     /// an opaque datum by the library.
-    pub fn channel_count(self) -> u8 {
+    pub fn channels_type(self) -> ChannelsType {
         match self {
             ExtendedColorType::L1 |
             ExtendedColorType::L2 |
             ExtendedColorType::L4 |
             ExtendedColorType::L8 |
             ExtendedColorType::L16 |
-            ExtendedColorType::Unknown(_) => 1,
+            ExtendedColorType::Unknown(_) => ChannelsType::L,
```

Comment:
This sets `ExtendedColorType::Unknown` to have `ChannnelsType::L`, I don't think that is quite right?

Reply:
According to the comment the only reasonable interpretation for an unknown color type absent other context is as a single channel. It's not quite as clear that the *only* reasonable interpretation of an unknown single channel image absent other context is luminance, but that's the assumption that's made here. However I do think it is *a* reasonable default interpretation. An alternative (if `ChannelsType` remains) is to make this return an `Option`, as you suggested.

Reply:
Would it also be possible to have a `ChannelsType` opaque?